ibis/backends/base/sql/alchemy/client.py

```diff
@@ -329,3 +329,105 @@ def _sqla_table_to_expr(self, table):
+        if_exists: Optional[str] = 'append',
+    ) -> None:
+        """
+        Insert the given data to a table in backend.
+
+        Parameters
+        ----------
+        to_table_name : string
+            name of the table to which data needs to be inserted
+        database : string, optional
+            name of the attached database that the table is located in.
+        data_obj : pd.Dataframe (Later can also be used for sending in
+        table_expr)
+            data_obj is the dataframe containing data which needs to be
+            inserted to to_table_name
+            (Later when table_expr related implementation is done this
+            variable can be used to take in that data)
+        from_table_name : string, optional
+            name of the table from which data needs to be inserted
+        if_exists : string, default 'append'
+            The values available are: {‘fail’, ‘replace’, ‘append’}
+
+        Raises
+        -------
+        ValueError
+            You must pass either data_obj (pandas Dataframe) or
+            from_table_name (table name to insert data from)
+
+        ValueError
+            Sorry, can't insert from both the data_obj (dataframe)
+            and the from_table_name (table). Please use only one
+            parameter.
+
+        ValueError
+            No operation is being performed. Either the data_obj
+            parameter is not a pandas dataframe or the
+            from_table_name parameter is not of string datatype.
+
+        """
+
+        if data_obj is None and from_table_name is None:
+            raise ValueError(
+                'You must pass either data_obj (pandas Dataframe)'
+                ' or from_table_name (table name to insert data from)'
+            )
+
+        if isinstance(data_obj, pd.DataFrame) and isinstance(
+            from_table_name, str
+        ):
+            raise ValueError(
+                "Sorry, can't insert from both the data_obj (dataframe)"
+                " and the from_table_name (table). Please use only one"
+                " parameter."
+            )
+
+        params = {}
+        if database is None:
+            params['schema'] = self.database_name
+            database = self.database_name
+
+        if isinstance(data_obj, pd.DataFrame):
+            data_obj.to_sql(
+                to_table_name,
+                self.con,
+                index=False,
+                if_exists=if_exists,
+                **params,
+            )
+        elif isinstance(from_table_name, str):
+            to_table_expr = self.table(to_table_name)
+            to_table_schema = to_table_expr.schema()
+
+            to_table = self._table_from_schema(
+                to_table_name,
+                to_table_schema,
+                database=database or self.current_database,
+            )
+
+            from_table_expr = self.table(from_table_name)
+
+            with self.begin() as bind:
+                to_table.create(bind=bind)
```

Comment:
Feels like we should be using `if_exists` here, and create, replace or fail depending on its value, right? Would be good to have tests for those too if not too complex.

Reply:
Firstly, `if_exists` must be used on `to_table` and not `from_table`.  
Coming to the usage part, I haven't used `if_exists` here because of a couple of reasons. One is, there's no parameter for it to be included in the `.table()` function so the backend is assuming the table will exist with the given name. Another reason is, I couldn't find functionality in the `.load_data()` and `.create_table()` functions which use another table (`from_table`) for its operations, and hence couldn't implement it similar to that. (This is an additional feature apart from using a dataframe for inserting). Earlier I was converting the `from_table` to a dataframe and then using `to_sql` for insertion which could accept the `if_exists` param and hence it looked fine.  
If you have any thoughts or sample reference which I can follow for the usage of the `if_exists` parameter by using the AlchemyClient operations I'll be happy to update this implementation.
Thanks!

Reply:
You simply need to do this manually. Something like:
```python
if if_exists == 'fail' and self.exists_table(to_table_name):
    raise RuntimeError('The table already exists')
elif if_exists == 'replace' and self.exists_table(to_table_name):
    self.drop(table_name_from)
...
```

Reply:
Updated.